geom/alg_distance_test.go

```diff
@@ -0,0 +1,142 @@
+		{"POINT EMPTY", "MULTIPOINT(0 0,EMPTY)", false, 0},
+		{"POINT(0 0)", "MULTIPOINT EMPTY", false, 0},
+		{"POINT(0 0)", "MULTIPOINT(EMPTY)", false, 0},
+		{"POINT(0 0)", "MULTIPOINT(0 0)", true, 0},
+		{"POINT(0 0)", "MULTIPOINT(0 1)", true, 1},
+		{"POINT(0 0)", "MULTIPOINT(0 1,1 0)", true, 1},
+		{"POINT(0 0)", "MULTIPOINT(0 2,3 0)", true, 2},
+		{"POINT(0 0)", "MULTIPOINT(0 2,1 1,3 0)", true, math.Sqrt(2)},
+
+		{"POINT EMPTY", "MULTILINESTRING EMPTY", false, 0},
+		{"POINT EMPTY", "MULTILINESTRING(EMPTY)", false, 0},
+		{"POINT EMPTY", "MULTILINESTRING(EMPTY,EMPTY)", false, 0},
+		{"POINT(0 0)", "MULTILINESTRING EMPTY", false, 0},
+		{"POINT(0 0)", "MULTILINESTRING(EMPTY)", false, 0},
+		{"POINT(0 0)", "MULTILINESTRING(EMPTY,EMPTY)", false, 0},
+		{"POINT EMPTY", "MULTILINESTRING((0 0,1 1))", false, 0},
+		{"POINT EMPTY", "MULTILINESTRING((0 0,1 1),EMPTY)", false, 0},
+		{"POINT(1 1)", "MULTILINESTRING((0 2,1 2),(10 0,10 1))", true, 1},
+		{"POINT(1 1)", "MULTILINESTRING((10 0,10 1),(0 2,1 2))", true, 1},
+
+		{"POINT EMPTY", "MULTIPOLYGON EMPTY", false, 0},
+		{"POINT EMPTY", "MULTIPOLYGON(EMPTY)", false, 0},
+		{"POINT(0 0)", "MULTIPOLYGON EMPTY", false, 0},
+		{"POINT(0 0)", "MULTIPOLYGON(EMPTY)", false, 0},
+		{"POINT EMPTY", "MULTIPOLYGON(((0 0,0 1,1 0,0 0)))", false, 0},
+		{"POINT EMPTY", "MULTIPOLYGON(((0 0,0 1,1 0,0 0)))", false, 0},
+		{"POINT(0 0)", "MULTIPOLYGON(((10 0,11 0,10 1,10 0)),((0 3,1 3,0 4,0 3)))", true, 3},
+		{"POINT(0 0)", "MULTIPOLYGON(((0 3,1 3,0 4,0 3)),((10 0,11 0,10 1,10 0)))", true, 3},
+
+		{"POINT EMPTY", "GEOMETRYCOLLECTION EMPTY", false, 0},
+		{"POINT EMPTY", "GEOMETRYCOLLECTION(POINT EMPTY)", false, 0},
+		{"POINT EMPTY", "GEOMETRYCOLLECTION(GEOMETRYCOLLECTION(POINT EMPTY))", false, 0},
+		{"POINT EMPTY", "GEOMETRYCOLLECTION(POINT(0 0))", false, 0},
+		{"POINT EMPTY", "GEOMETRYCOLLECTION(GEOMETRYCOLLECTION(POINT(0 0)))", false, 0},
+		{"POINT(0 0)", "GEOMETRYCOLLECTION EMPTY", false, 0},
+		{"POINT(0 0)", "GEOMETRYCOLLECTION(POINT EMPTY)", false, 0},
+		{"POINT(0 0)", "GEOMETRYCOLLECTION(GEOMETRYCOLLECTION(POINT EMPTY))", false, 0},
+		{"POINT(0 0)", "GEOMETRYCOLLECTION(POINT(1 1))", true, math.Sqrt(2)},
+		{"POINT(0 0)", "GEOMETRYCOLLECTION(GEOMETRYCOLLECTION(POINT(1 1)))", true, math.Sqrt(2)},
+
+		{"LINESTRING EMPTY", "LINESTRING EMPTY", false, 0},
+		{"LINESTRING EMPTY", "LINESTRING(0 0,1 1)", false, 0},
+		{"LINESTRING(0 0,1 1)", "LINESTRING EMPTY", false, 0},
+		{"LINESTRING(0 0,4 0)", "LINESTRING(1 1,3 1)", true, 1},
+		{"LINESTRING(0 0,4 0)", "LINESTRING(1 1,2 2)", true, 1},
+		{"LINESTRING(0 0,4 0)", "LINESTRING(3 1,2 2)", true, 1},
+		{"LINESTRING(0 0,4 0)", "LINESTRING(-1 1,-1 2)", true, math.Sqrt(2)},
+		{"LINESTRING(0 0,4 0)", "LINESTRING(5 1,5 2)", true, math.Sqrt(2)},
+
+		{"POLYGON((1 2,0 1,1 0,2 1,1 2))", "POLYGON((4 2,3 1,4 0,5 1,4 2))", true, 1},
+		{"MULTIPOLYGON(((1 2,0 1,1 0,2 1,1 2)),((1 5,0 4,1 3,2 4,1 5)))", "POLYGON((4 2,3 1,4 0,5 1,4 2))", true, 1},
+	} {
```

Comment:
I can't immediately see if you have both concave and convex polygons being tested here, but I think that might be a good idea.